### ISSUE TITLE
allow confetti to be rendered by OffscreenCanvas

### ISSUE THREAD

**Issue by catdad** [catdad/canvas-confetti#83]
https://developers.google.com/web/updates/2018/08/offscreen-canvas#unblock_main_thread

- [x] the global `confetti` method should always use a worker when it is capable of doing so without the user having to do anything
- [x] a manually created instance of `confetti` should have an option controlling whether workers may be used, defaults to `false`

**Comment by catdad** [comment]
This is how React sets up a web worker without being able to access a server-side URL:

```javascript
export default class WebWorker {
	constructor(worker) {
		const code = worker.toString();
		const blob = new Blob(['('+code+')()']);
		return new Worker(URL.createObjectURL(blob));
	}
}
```

**Comment by catdad** [comment]
Published in 1.0.0
